### ISSUE TITLE
[Issue] Live TV Playback Fails With Tvheadend if Transcode Path Contains Trailing Slash

### ISSUE THREAD

**Issue by i3Craig** [jellyfin/jellyfin#10299]
**Describe The Bug**

If the transcode path in Jellyfin playback config is set to the following path `/dev/shm/jellyfin/` (and the folder exists, has the correct permissions and has available space), live TV will fail to playback with the following error:
```
Playback Error
This client isn't compatible with the media and the server isn't sending a compatible media format.
```
Note that recordings work correctly even with the "invalid" transcoding path.

**Steps To Reproduce**

1. Setup a folder for transcoding on the Jellyfin server. Ensure this has the correct permissions. In my example, `sudo mkdir /dev/shm/jellyfin; sudo chown jellyfin:jellyfin /dev/shm/jellyfin`.
2. Go to the Menu bar > Dashboard > select Playback
3. Find the Transcode path and enter the full path to the folder created in (1) with a trailing slash. In my example, enter '/dev/shm/jellyfin/`.
4. Try to watch any live TV through the TVHeadend plugin. The error listed above ("Playback Error") will be displayed.

**Expected Behavior**

Jellyfin should playback the live TV without throwing an error.

**Logs**

Below are the relevant logs where Jellyfin sends `ffmpeg` a `q` to stop the transcoding.
```
[2023-09-25 20:16:02.870 -05:00] [INF] [27] Jellyfin.Api.Helpers.TranscodingJobHelper: Stopping ffmpeg process with q command for "/dev/shm/jellyfin/a566d30c25488ca5c002bb7a912073d4.m3u8"
[2023-09-25 20:16:03.180 -05:00] [INF] [27] Jellyfin.Api.Helpers.TranscodingJobHelper: FFmpeg exited with code 0
[2023-09-25 20:16:03.181 -05:00] [INF] [27] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/dev/shm/jellyfin/a566d30c25488ca5c002bb7a912073d4.m3u8"

```

**Screenshots**

**System (please complete the following information):**
 - Platform: Linux (Arch X86-64)
 - Browser: Chrome and Firefox show the same error.
 - Jellyfin Version: 10.8.11

**Additional Context**

After digging through the relevant code in `jellyfin` (not `jellyfin-web`), I found the code that triggered the "Invalid segment" response that the server sent, which caused the client to fail to playback the live TV, which stopped playback and caused the server to stop `ffmpeg`. 
Around line 68 of `jellyfin/Jellyfin.Api/Controllers/HlsSegmentController.cs`, we find the following code:
```cs
    public ActionResult GetHlsAudioSegmentLegacy([FromRoute, Required] string itemId, [FromRoute, Required] string segmentId)
    {
        // TODO: Deprecate with new iOS app
        var file = segmentId + Path.GetExtension(Request.Path);
        var transcodePath = _serverConfigurationManager.GetTranscodePath();
        file = Path.GetFullPath(Path.Combine(transcodePath, file));
        var fileDir = Path.GetDirectoryName(file);
        if (string.IsNullOrEmpty(fileDir) || !fileDir.StartsWith(transcodePath, StringComparison.InvariantCulture))
        {
            return BadRequest("Invalid segment.");
        }

        return FileStreamResponseHelpers.GetStaticFileResult(file, MimeTypes.GetMimeType(file));
    }
```

It turns out that the path sent to the server from the client when requesting a new mpegts segment is directly compared with the configured transcode path. Moreover, after adding some logging, I found that `fileDir` contains the requested path to the mpegts file, but does not contain a trailing slash. Thus, the StartsWith returns `false`, causing the method to return with "Invalid segment.". I updated the server configuration Transcode Path described earlier to not include the trailing slash (so it is now set to "/dev/shm/jellyfin") and live TV started working without any errors (playback recordings also works without the trailing slash, but it worked with the trailing slash for some reason).

**Conclusion**
My idea is that maybe, if possible, a warning could be shown or the configuration could be visualy flagged if a user attempts to enter a transcode path that does contain a trailing slash. I know the file browser won't leave a trailing slash by default, but if a user copy-pastes the transcode path, this can happen. Given that the code assumes no trailing slash (and that this can lead to hard-to-debug issues), I think the safest option would be to make the textbox red and disable the save configuration button if the user tries to enter an invalid transcode path with a trailing slash.

Note that I haven't tested this on Windows and it is possible that the trailing slash is needed there, but I am not sure.

**Comment by i3Craig** [comment]
This is still an issue in Jelly yfin 10.8.13
![image](https://github.com/jellyfin/jellyfin/assets/70919757/63435157-c86f-4e99-86b6-de42386e66b8)

**Comment by jedthe3rd** [comment]
I ran into this issue as well. Adding a trailing slash to my custom transcode path breaks custom m3u streams from playing properly. In my case I am running on Windows 11.

**Comment by EmPiFree** [comment]
Error still persists on my side: 
![image](https://github.com/jellyfin/jellyfin/assets/22915116/911f3500-d696-4b8a-ad45-5e4ced168b0e)

**Comment by garymoon** [comment]
Removing the trailing slash from the end of my custom transcode path (`Dashboard -> Playback -> Transcoding -> Transcode path`) fixed live playback for me on v10.10.1. Thanks everyone who chimed in! 💙
